corehq/ex-submodules/casexml/apps/phone/fixtures.py

```diff
@@ -26,31 +36,49 @@ class FixtureGenerator(object):
     """
 
     def __init__(self):
-        self._generator_funcs = {}
+        def to_provider(provider_id, func_path):
+            func = to_function(func_path)
+            return FixtureProvider(id=provider_id, func=func) if func else None
+
+        self._generator_providers = {}
         if hasattr(settings, "FIXTURE_GENERATORS"):
-            for group, func_paths in settings.FIXTURE_GENERATORS.items():
-                self._generator_funcs[group] = filter(None, [
-                    to_function(func_path) for func_path in func_paths
+            for group, func_tuple in settings.FIXTURE_GENERATORS.items():
+                self._generator_providers[group] = filter(None, [
+                    to_provider(provider_id, func_path) for provider_id, func_path in func_tuple
                 ])
 
-    def get_fixtures(self, user, version, last_sync=None, group=None):
-        """
-        Gets all fixtures associated with an OTA restore operation
-        """
+    def _get_fixtures(self, group, fixture_id_or_prefix, user, version, last_sync):
         if version == V1:
             return []  # V1 phones will never use or want fixtures
 
-        funcs = []
         if getattr(user, "_hq_user", False):
             user = user._hq_user
-        if isinstance(user, CommCareUser):
-            if group:
-                funcs = self._generator_funcs.get(group, [])
-            else:
-                funcs = itertools.chain(*self._generator_funcs.values())
-
-        return itertools.chain(*[func(user, version, last_sync)
-                                 for func in funcs])
+
+        if not isinstance(user, CommCareUser):
+            return []
+
+        if group:
+            providers = self._generator_providers.get(group, [])
+        else:
+            providers = itertools.chain(*self._generator_providers.values())
+
+        if fixture_id_or_prefix:
+            providers = [provider for provider in providers if provider.id.startswith(fixture_id_or_prefix)]
```

Comment:
Need to use `startswith` here since some fixtures have dynamic ID's e.g. `item-list:{tag_name}`